### ISSUE TITLE
Port internal python scripts to python3

### ISSUE THREAD

**Issue by ndufresne** [harfbuzz/harfbuzz#1501]
Currently building harfbuzz seems to require python2 (shebang of these scripts are all ```#!/usr/bin/env python``` and some script requires some small changes according to 2to3. Python2 is getting old and we are trying to not depends on having this tool in our build scripts in GStreamer project.

So this ticket is to basically project maintainer(s) are ready to drop python2 support in favour of python3. The porting is relatively trivial scripted, so I can provide an PR for that.

**Comment by behdad** [comment]
I'm fairly sure they run with python3.  Did you actually face an error, or just judging from 2to3 output?

**Comment by ndufresne** [comment]
For some reason, only gen-def.py for cross-windows build was called, and this one works just fine in python3. I probably need to enable more options to get the other scripts to be called. So yes, this is judging from 2to3. Clearly the __future__ get ignored properly, there is some list() casting that 2to3 seems to suggest that may be required.

All at the same time, Nirbheek was telling me that someone submited a meson port of the build system, which only works on python3.

**Comment by ebraminio** [comment]
I did most of py3 compatibility fixes and now our `archlinux-debug-O0-py3` and Windows-msys2 bots https://circleci.com/gh/harfbuzz/harfbuzz/68490 are running python3, but there are corner cases may are not python3 compatible. I am personally OK to make the scripts explicitly dependent to py3, maybe however Behdad is not and anyway we can wait for distro to make the switch. We don't like to make explicit dependency to any non-default Python packages so we like to keep 2to3 shims locally, which is as now. Closing this but we welcome any PR related to the topic.
